server/events/yaml/valid/global_cfg_test.go

```diff
@@ -592,6 +598,19 @@ func TestRepo_IDString(t *testing.T) {
 	Equals(t, "/regex.*/", (valid.Repo{IDRegex: regexp.MustCompile("regex.*")}).IDString())
 }
 
+func TestRepo_BranchMatches(t *testing.T) {
+	// Test matches when no branch regex is set.
+	Equals(t, true, (valid.Repo{}).BranchMatches("main"))
+
+	// Test regexes.
+	Equals(t, true, (valid.Repo{BranchRegex: regexp.MustCompile(".*")}).BranchMatches("main"))
+	Equals(t, true, (valid.Repo{BranchRegex: regexp.MustCompile("main")}).BranchMatches("main"))
```

Comment:
Consider adding test simillar to the line above (where input is plain branch name and not reg-ex) but when the match actually returs false (the names does not match).

Reply:
Added some cases for `^master$`, still sticking with branch is always a regex for now.